test/deterministicmnlist/SimplifiedMNListDiff.js

```diff
@@ -1,45 +1,58 @@
   "deletedMNs": [
-    '01040eb32f760490054543356cff463865633439dd073cffa570305eb086f70a'
+    "14d924611e20307338c0937ad746226c0b50b01d47824c9ef08e141cc4635c9f"
   ],
   "mnList": [
     {
-      "proRegTxHash": "01040eb32f760490054543356cff463865633439dd073cffa570305eb086f70e",
-      "service": "[0:0:0:0:0:0:0:0]:0",
-      "keyIDOperator": "0000000000000000000000000000000000000000",
-      "keyIDVoting": "c2ae01fb4084cbc3bc31e7f59b36be228a320404",
+      "proRegTxHash": "fef106ff6420f9c6638c9676988a8fc655750caafb506c98cb5ff3d4fea99a41",
+      "confirmedHash": "0000000005d5635228f113b50fb5ad66995a7476ed20374e6e159f1f9e62347b",
+      "service": "45.48.177.222:19999",
+      "pubKeyOperator": "842476e8d82327adfb9b617a7ac3f62868946c0c4b6b0e365747cfb8825b8b79ba0eb1fa62e8583ae7102f59bf70c7c7",
+      "keyIDVoting": "ca58159731cf7e3791958050d16bce02a64223ce",
+      "isValid": true
+    },
+    {
+      "proRegTxHash": "a4d877cee62f82868034fb678436d87afbb13330d2b66a24ae1d357f0de55c68",
+      "confirmedHash": "00000000069c41d7444a7da5d67f222224e9e37590c474f102ee1ae0da998f39",
+      "service": "83.80.229.213:19999",
+      "pubKeyOperator": "16415af54406658be9ea44d82b6b502bb90d93e32997484533a8a71a4ed98d12cea3709d84a5835b6ad8ed48d3101633",
+      "keyIDVoting": "c0f278bb9cd78a55d96553c080ac069f929e66d0",
       "isValid": false
     },
     {
-      "proRegTxHash": "f7737beb39779971e9bc59632243e13fc5fc9ada93b69bf48c2d4c463296cd5a",
-      "service": "207.154.244.13:19999",
-      "keyIDOperator": "03d90b1cdc04f1dbe435a4ba51ca2d1ddb53e08c",
-      "keyIDVoting": "43ce12751c4ba45dcdfe2c16cefd61461e17a54d",
-      "isValid": true
+      "proRegTxHash": "a690051e69de6e36eeba664bff34e017f973d27ce91c1f2247120e8ce586b1f1",
+      "confirmedHash": "0000000019091441469a98f9a8889d94e54723286fe1cd13703aa6b652fc4863",
+      "service": "149.248.55.77:19999",
+      "pubKeyOperator": "8b165f653a3970a17f432f6c3abb8b681c71a3775f998fff322341d2994767c167c8a43b1b4661b9c01ef637763d4d81",
+      "keyIDVoting": "5cb8fde05268ac28619b483b2bff8e554b41314d",
+      "isValid": false
     },
     {
-      "proRegTxHash": "75aa128db4cd7679fd88206bd6ef71f57e1b6fe04c2da5515193a6fcd40a47eb",
-      "service": "159.89.110.184:19999",
-      "keyIDOperator": "03d90b1cdc04f1dbe435a4ba51ca2d1ddb53e08c",
-      "keyIDVoting": "03d90b1cdc04f1dbe435a4ba51ca2d1ddb53e08c",
+      "proRegTxHash": "6a46c1a01ee6ed517ff35ecb9cdffde3d1d0a5bb89e8c05e916a204134d7909f",
+      "confirmedHash": "00000000092935037c1cd00b67dbc09deb49f8853b101598a6b4ad1dbc177afa",
+      "service": "51.38.112.99:19999",
+      "pubKeyOperator": "88ee437bc0ba444b71a6b8a525146e9c748a8430fc85ad47beda04bb2e5b698bca9f3d5a5d5dfdd5990cd08daa07371f",
+      "keyIDVoting": "e44ac3724646e0eab85df1b075ce81976711bb8c",
       "isValid": true
     }
-  ]
+  ],
+  "merkleRootMNList": "f3ce92939573af9f8c8717ca775eb0d7a0b96084cec93c573d3e49a169cb8e9e"
 };
 
```

Comment:
I think there should be a DIP4 test somewhere that checks if calculating the merkleRootMNList according to the rules described in DIP4 results in the hash of the merkleRootMNList. Something like this we have in dash-spv tests which is not working yet

```
it('should validate correct mnlistmerkleroot', () => {
  // not working yet
  dash-spv.merkleProofs.validateMnListMerkleRoot(mnListDiffJSON.merkleRootMNList, mnListDiffJSON.mnList)
    .should.equal(true);
});
```

Reply:
but we can do  this test also after moving dash-spv to dashcore-lib

Reply:
@Cofresi we can't calculate it here. It's a root for the whole list, not just for this diff. You can test it by running: 
`./dash-cli -testnet protx diff 1000 21000`
`./dash-cli -testnet protx diff 2100 21179`
Those two commands will contain the same root, but diffs will be different.

Initially, I wanted to add construction of the full list from diffs in another PR, as EV-1009 become too big. I want to add it a couple PRs later: first of all, I want to move the Merkle root calculation from MerkleBlock class to utils, and afterward to create an MNList class, which would have such methods as `applyDiff` and `verify`. The last would also include root check